### PR TITLE
Upgrade support version 26.1.0 -> 27.0.2

### PR DESCRIPTION
**Pull request by repitch**
- Upgrade support version 26.1.0 -> 27.0.2
- Upgrade compile sdk version 26 -> 27
- Fix accompanying bugs
- Get rid of legacy compat methods
- Move lib versions inside base gradle file

### CODE REVIEW

**Comment by repitch** [comment]
Here's what changed on CoordinatorLayout from 26.10 to 27.0.2: [https://www.diffchecker.com/SEj02bXa](https://www.diffchecker.com/SEj02bXa)
The difference is: `resetTouchBehaviors() -> resetTouchBehaviors(boolean notifyOnInterceptTouchEvent)`, so `resetTouchBehaviors` can send `ACTION_CANCEL` to `behavior.onInterceptTouchEvent` if `notifyOnInterceptTouchEvent==true`.
`resetTouchBehaviors(true)` is called inside `CoordinatorLayout.onInterceptTouchEvent(..)` when action is ACTION_DOWN, ACTION_UP or ACTION_CANCEL.
So, after ACTION_DOWN, inside `BottomSheetBehaviorGoogleMapsLike#onInterceptTouchEvent()` this new logic cause calling `mScrollVelocityTracker.clear();` and velocity is missing.
Summing up, our coordinator's child (behavior) inside `onInterceptTouchEvent` can't differ ACTION_CANCEL from ACTION_DOWN and ACTION_UP in some cases, so velocity clearing is now called only on ACTION_DOWN (before any gestures).

**Review by seva91 — APPROVED** [review]

**Review by tralfamadore — APPROVED** [review]

**Review by MaximKorobovAtTinkoff — APPROVED** [review]
Brilliant.
